manim/mobject/coordinate_systems.py

```diff
@@ -509,15 +579,23 @@ class NumberPlane(Axes):
 
     def __init__(
         self,
-        x_range=(-config["frame_x_radius"], config["frame_x_radius"], 1),
-        y_range=(-config["frame_y_radius"], config["frame_y_radius"], 1),
-        x_length=None,
-        y_length=None,
-        axis_config=None,
-        y_axis_config=None,
-        background_line_style=None,
-        faded_line_style=None,
-        faded_line_ratio=1,
+        x_range: Optional[Union[List[float], np.ndarray]] = (
+            -config["frame_x_radius"],
+            config["frame_x_radius"],
+            1,
+        ),
+        y_range: Optional[Union[List[float], np.ndarray]] = (
+            -config["frame_y_radius"],
+            config["frame_y_radius"],
+            1,
+        ),
+        x_length: Optional[float] = None,
+        y_length: Optional[float] = None,
+        axis_config: Optional[dict] = None,
+        y_axis_config: Optional[dict] = None,
+        background_line_style: Optional[dict] = None,
+        faded_line_style: Optional[dict] = None,
+        faded_line_ratio: Optional[float] = 1,
```

Comment:
`faded_line_ratio` only produces expected results when int
```suggestion
        faded_line_ratio: Optional[int] = 1,
```